alpha/apps/kaltura/lib/batch2/kFlowHelper.php

```diff
@@ -2967,6 +2967,49 @@ public static function addDeliveryTagToEntry(entry $entry)
 			$entry->save();
 		}
 	}
+
+	public static function handleDelayedNotification($object)
+	{
+		$delayedLockedJobs = [];
+		switch (true)
+		{
+			case ($object instanceof entry):
+			{
+				$delayedLockedJobs = BatchJobLockPeer::retrieveByEntryIdAndStatus($object->getEntryId(), BatchJob::BATCHJOB_STATUS_DELAYED);
+				break;
+			}
+		}
+
+		if (count($delayedLockedJobs) > 0)
+		{
+			foreach ($delayedLockedJobs as $job)
+			{
+				/* @var $job BatchJobLock */
+				$delayedJob = BatchJobPeer::retrieveByPK($job->getId());
+				/* @var $job BatchJob */
+				if ($delayedJob)
+				{
+					$jobData = $delayedJob->getData();
+					if ($jobData)
```

Comment:
let make sure that this job is of type notification since we need the jobData to be kEventNotificationDispatchJobData and we will access $jobData->getEventDelayedConditions()

Reply:
not sure i understand

Reply:
you men doing this?
if ($jobData && $jobData instanceof kEventNotificationDispatchJobData)